rewrite-maven/src/main/java/org/openrewrite/maven/ChangeDependencyGroupIdAndArtifactId.java

```diff
@@ -146,18 +147,28 @@ public Validated<Object> validate() {
 
     @Override
     public TreeVisitor<?, ExecutionContext> getVisitor() {
-
         return new MavenVisitor<ExecutionContext>() {
             @Nullable
             final VersionComparator versionComparator = newVersion != null ? Semver.validate(newVersion, versionPattern).getValue() : null;
             @Nullable
             private Collection<String> availableVersions;
 
             @Override
-            public Xml visitTag(Xml.Tag tag, ExecutionContext ctx) {
+            public Xml visitDocument(Xml.Document document, ExecutionContext ctx) {
+                // Any managed dependency change is unlikely to use the same version, so only update selectively.
+                if ((changeManagedDependency == null || changeManagedDependency) && newVersion != null || versionPattern != null) {
```

Comment:
This seemed a reasonable limit to impose, as the recipe itself also has a `if (newVersion != null)`.